### PR TITLE
Covid-19 Unique data

### PR DESCRIPTION
**Pull request by oyelakinG9**
# COVID_19 UNIQUE STAT 
> Project Requirement
- Select the API you are going to use.
- Build files structure for your React app.
- Prepare routes and navigation in your app.

- Make sure that a user can display a list of items and filter them. Data should come from the API.
- Create the tests for the application.

- Style your components to match the design provided.
- Deploy the project and test for final details.
- Record a video for your project.
- Create a good README and PR description.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,40 @@
 ![](https://img.shields.io/badge/Microverse-blueviolet)
 
-# Project Name
+# COVID-19 UNIQUE DATA
+> Unique data is a react project that provides COVID-19 data for each country. It is a SPA website provides users with an immediate update on COVID-19 data. It showcases my work at the end of the react milestone.
```
Comment (MrBrN197) [inline README.md]: - [ ] please ensure to add an acknowledgments section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: README.md [hunk 2]
```diff
@@ -36,6 +41,18 @@
...
+
+- Major languages
+- Frameworks
+- Technologies used
+
+## Live Demo (if available)
+
+[Live Demo Link](https://livedemo.com)[Not available]
+
+
```
Comment (MrBrN197) [inline README.md]: - [ ] your project functionality is working well on localhost, although please ensure to deploy this project using either Heroku, Netlify, or Github pages.

File: src/pages/CountriesList.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    <Nav />
+    <Link to="/about" className="arrow">
+      {' '}
+      <FontAwesomeIcon icon={faAngleDoubleLeft} className="mx-1" />
+      About
+    </Link>
+    <div className="page-title">
+      <p data-content="Countries where COVID-19 has spread">Countries where COVID-19 has spread</p>
+    </div>
+    {/* Counties list */}
```
Comment (MrBrN197) [inline src/pages/CountriesList.js:19]:
```suggestion
```

File: src/components/countries/Country.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/prop-types */
```
Comment (MrBrN197) [inline src/components/countries/Country.js]: - [ ] please kindly avoid using several eslint-disable comments to get around ESlint errors. these ESlint errors are at best showing that you are not adhering to some coding standards or at worst highlighting a possible underlying issue with your code. please kindly address these issues.

File: src/components/regions/Region.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable react/prop-types */
```
Comment (MrBrN197) [inline src/components/regions/Region.js]: - [ ] please kindly avoid using several eslint-disable comments to get around ESlint errors. these ESlint errors are at best showing that you are not adhering to some coding standards or at worst highlighting a possible underlying issue with your code. please kindly address these issues.

File: src/components/regions/Regions.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  const { countries, loaded } = useSelector((state) => state.countries, shallowEqual);
+  const [regions, setRegions] = useState(null);
+  const [activePage, setActivePage] = useState(1);
+
+  useEffect(() => {
+    dispatch(fetchCountriesAction());
+    if (loaded) {
+      const countryObj = countries.filter((obj) => obj.id === id);
+      // Non camelCase are needed here since they're used by the Narrativa API.
+      /* eslint-disable camelcase */
```
Comment (MrBrN197) [inline src/components/regions/Regions.js:22]: - [ ] please kindly avoid using several eslint-disable comments to get around ESlint errors. these ESlint errors are at best showing that you are not adhering to some coding standards or at worst highlighting a possible underlying issue with your code. please kindly address these issues.
- [ ] in these case you rename the variables that you get from the Narrativa API to have camelCase

File: src/utils/countries-api.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable consistent-return */
```
Comment (MrBrN197) [inline src/utils/countries-api.js]: - [ ] please kindly avoid using several eslint-disable comments to get around ESlint errors. these ESlint errors are at best showing that you are not adhering to some coding standards or at worst highlighting a possible underlying issue with your code. please kindly address these issues.

File: src/utils/filterCountries-api.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable consistent-return */
```
Comment (MrBrN197) [inline src/utils/filterCountries-api.js]: - [ ] please kindly avoid using several eslint-disable comments to get around ESlint errors. these ESlint errors are at best showing that you are not adhering to some coding standards or at worst highlighting a possible underlying issue with your code. please kindly address these issues.

File: src/utils/filterCountries-api.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+const fetchCountries = async (name) => {
+  const res = await axios(config);
+  if (res.status === 200) {
+    const contriesData = await res.data.dates[`${todayDate}`].countries;
+    const allCountries = Object.keys(contriesData).map((key) => contriesData[key]);
+    const filteredCountry = allCountries.filter((obj) => obj.id.includes(name));
+
+    // Non camelCase are needed here since they're used by the Narrativa API.
+
+    /* eslint-disable camelcase */
```
Comment (MrBrN197) [inline src/utils/filterCountries-api.js:26]: - [ ] please kindly avoid using several eslint-disable comments to get around ESlint errors. these ESlint errors are at best showing that you are not adhering to some coding standards or at worst highlighting a possible underlying issue with your code. please kindly address these issues.

File: src/utils/countries-api.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+const fetchCountry = async () => {
+  const res = await axios(config);
+  if (res.status === 200) {
+    const contriesData = await res.data.dates[`${todayDate}`].countries;
+    const allCountries = Object.keys(contriesData).map((key) => contriesData[key]);
+    const filteredCountries = allCountries.filter((obj) => europecountries.includes(obj.id));
+
+    // Non camelCase are needed here since they're used by the Narrativa API.
+
+    /* eslint-disable camelcase */
```
Comment (MrBrN197) [inline src/utils/countries-api.js:28]: - [ ] please kindly avoid using several eslint-disable comments to get around ESlint errors. these ESlint errors are at best showing that you are not adhering to some coding standards or at worst highlighting a possible underlying issue with your code. please kindly address these issues.

**Review by MrBrN197 — CHANGES_REQUESTED** [review]
## Hi, @oyelakinG9  👋 

You've done a really nice job so far.
There are some issues that you still need to work on to go to the next project but you are almost there!
## To highlight 👏👏
- ✔ Great work creating an entire about page with details about the project. 👏

## Required Changes ♻️
- [ ] In your video please make sure to highlight a piece of code that is particularly interesting.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (oyelakinG9) [reply README.md]: Already deployed but due to the result of a merge conflict. It takes time to show. It's there now. Thanks

File: src/pages/CountriesList.js (hunk 3 above)
Comment (oyelakinG9) [reply src/pages/CountriesList.js:19]: Removed now. Thanks

File: src/utils/countries-api.js (hunk 10 above)
Comment (oyelakinG9) [reply src/utils/countries-api.js:28]: That's just the best I can do for now. I've tried to correct it.

File: src/components/countries/Country.js (hunk 4 above)
Comment (oyelakinG9) [reply src/components/countries/Country.js]: Thanks @MrBrN197 
I've removed it.

File: README.md (hunk 2 above)
Comment (rloterh) [reply README.md]: This issue is well resolved! ✔️

File: src/pages/CountriesList.js (hunk 3 above)
Comment (rloterh) [inline src/pages/CountriesList.js:19]: - [ ] Please remove this comment as requested.

File: src/components/countries/Country.js (hunk 4 above)
Comment (rloterh) [reply src/components/countries/Country.js]: This issue is well resolved! ✔️

File: src/components/regions/Regions.js (hunk 6 above)
Comment (rloterh) [inline src/components/regions/Regions.js:22]: - [ ] You have **disabled linter check** on your code. Please enable linter check on your code to ensure that your code is robust code and clean as recommended.

File: src/utils/countries-api.js (hunk 7 above)
Comment (rloterh) [reply src/utils/countries-api.js]: - [ ] You have **disabled linter check** on your code. Please enable linter check on your code to ensure that your code is robust code and clean as recommended.

File: src/utils/countries-api.js (hunk 10 above)
Comment (rloterh) [reply src/utils/countries-api.js:28]: - [ ] You have **disabled linter check** on your code. Please enable linter check on your code to ensure that your code is robust code and clean as recommended.

File: src/utils/filterCountries-api.js (hunk 8 above)
Comment (rloterh) [reply src/utils/filterCountries-api.js]: - [x] You have **disabled linter check** on your code. Please enable linter check on your code to ensure that your code is robust code and clean as recommended.

File: src/utils/filterCountries-api.js (hunk 9 above)
Comment (rloterh) [reply src/utils/filterCountries-api.js:26]: - [ ] You have **disabled linter check** on your code. Please enable linter check on your code to ensure that your code is robust code and clean as recommended.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Oyelaking, 

You did great on this project, and for making the changes requested in the previous review(s)! 👏👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- No linter errors

## Required Changes ♻️
_Kindly check the comments below**. Please take your time to carefully improve aspects outlined in the previous code review(s), and please check to see that the changes implemented are working as intended._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 3 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by oyelakinG9** [comment]
Thanks @MrBrN197. 
I can't remove it because it won't sensible with API. If check the API documentation No camelCase are used

File: src/components/regions/Regions.js (hunk 6 above)
Comment (oyelakinG9) [reply src/components/regions/Regions.js:22]: If I remove it it won't response to the API since No camelCase are used.

Comment (oyelakinG9) [reply src/components/regions/Regions.js:22]: It's for the API call, No camelCase are used

File: src/components/regions/Regions.js [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+      const regionsData = countryObj[0].regions.map(({
+        id, name, today_new_confirmed, today_new_deaths, date,
+      }) => ({
+        date,
+        id,
+        name,
+        todayNewConfirmed: today_new_confirmed || 0,
+        todayNewDeaths: today_new_deaths || 0,
```
Comment (MrBrN197) [inline src/components/regions/Regions.js:30]: You can alleivate the eslint camelCase error by doing this; renaming the variables as you destructure
```suggestion
      const regionsData = countryObj[0].regions.map(({
        id, name, today_new_confirmed: todayNewConfirmed, today_new_deaths: todayNewDeaths, date,
      }) => ({
        date,
        id,
        name,
        todayNewConfirmed: todayNewConfirmed || 0,
        todayNewDeaths: todayNewDeaths || 0,
```

**Comment by MrBrN197** [comment]
Hi, @oyelakinG9  I understand this but all it simply needs is for you to rename those variables as you destructure them. look at the suggestion I left here https://github.com/oyelakinG9/unique-data/pull/1#discussion_r825329360

although since you've removed a lot of the previous linter checks I think it's valid for these to be OPTIONAL now. so the next reviewer should approve the project.

File: src/components/regions/Regions.js (hunk 6 above)
Comment (MrBrN197) [reply src/components/regions/Regions.js:22]: look at these suggestion I've left https://github.com/oyelakinG9/unique-data/pull/1#discussion_r825329360

Comment (MrBrN197) [reply src/components/regions/Regions.js:22]: look at this suggestion I left https://github.com/oyelakinG9/unique-data/pull/1#discussion_r825329360

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Oyelaking, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by rukundoeric** [comment]
Hi @oyelakinG9,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------

**Pull request merged**